ls1intum/Artemis
# `General`: Improve usability of LTI launch

### Checklist
#### General
#### Client

### Motivation and Context
Improve the QoL of an LTI launch by prefilling the username field with the corresponding username for the LTI launch. That way, users can more easily now what user they are logging in as and only need to fill in the password.

Also prevents user from using ESC to close the initializer modal showing the password, forcing them to actually read the content and save the password. Adds a copy button for the password as well

### Description

- Prefills the username field with the corresponding username from the LTI launch
- Prevent users from closing modal with ESC (keyboard: false)
- Add copy button for password field

### Steps for Testing
Prerequisites:
- 1 Instructor

Launch as new LTI student 
1. Verify that password copy button is present in the password modal and that it works
2. Verify that it's not possible to close modal with "ESC" key (or any other key)

Launch as existing LTI student 
1. Verify that notification is shown "LTI launch as {username}" with the correct username
2. Verify that the username is already prefilled in the login bar



### Review Progress

#### Code Review
#### Manual Tests
### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| account.service.ts | 87% | 88% |
| user-route-access.service.ts | 78% | 89% |

### Screenshots

Password initializer modal with copy button

![image](https://user-images.githubusercontent.com/11006274/219600549-a3c925c3-6eb4-4a4a-afc4-7ef0a9f669d6.png)
